.github/workflows/ci.yml

```diff
@@ -45,3 +45,29 @@ jobs:
         run: pushd examples; ./build_all.sh || exit; popd
       - name: ci-debug-build
         run: pushd examples/blink; make debug RAM_START=0x20004000 FLASH_INIT=0x30051 || exit; popd
+
+  # Install compiler versions from the oldest current Ubuntu LTS
+  ci-legacy-compilers:
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+    # The type of runner that the job will run on
+    runs-on: ${{ matrix.os }}
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - uses: carlosperate/arm-none-eabi-gcc-action@v1
+        with:
+          release: '10.3-2021.07'
+      - run: arm-none-eabi-gcc --version
+      - name: setup-riscv-toolchain
+        run: sudo apt-get install -y gcc-riscv64-unknown-elf
+      - name: riscv-toolchain-version
+        run: riscv64-unknown-elf-gcc --version
+      #- name: ci-build
+      #  run: pushd examples; ./build_all.sh || exit; popd
+      - name: ci-debug-build
+        run: pushd examples/blink; make debug RAM_START=0x20004000 FLASH_INIT=0x30051 || exit; popd
```

Comment:
(I must have forgotten to hit the button on this comment) Could we split this out to a separate PR? I _think_ it's separate from the rest of the changes here.